Builds/CMake/CMakeFuncs.cmake

```diff
@@ -482,6 +482,15 @@ macro(setup_build_boilerplate)
 
   if (is_gcc)
     add_compile_options(-Wno-unused-but-set-variable -Wno-deprecated)
+
+    # use gold linker if available
+    execute_process(
+      COMMAND ${CMAKE_C_COMPILER} -fuse-ld=gold -Wl,--version
```

Comment:
I notice here you are using the C compiler here and in scons you are using the CXX compiler. I don't think it makes any practical difference, but perhaps stick with CXX for consistency?